### PR TITLE
ARC-658: Delete dev info data when repo is deleted from GitHub.

### PR DESCRIPTION
**Pull request by atrigueiro**
https://developer.atlassian.com/cloud/jira/software/rest/api-group-development-information/#api-devinfo-0-10-repository-repositoryid-delete

### CODE REVIEW

File: app.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ default_events:
 - pull_request_review_comment
 - push
 # - release
-# - repository
+- repository
```
Comment (KAbakumovAtl) [inline app.yml:43]: Would it require a new permissions from users?

**Review by KAbakumovAtl — APPROVED** [review]
Nice. Simple and elegant solution

File: src/github/index.ts [hunk 2]
```diff
@@ -54,4 +55,6 @@ export default (robot: Application) => {
 
 	robot.on("create", middleware(createBranch));
 	robot.on("delete", middleware(deleteBranch));
+
+	robot.on("repository.deleted", middleware(deleteRepository));
```
Comment (rachellerathbone) [inline src/github/index.ts:59]: I just tested deleting a repository and the webhook event I saw was `installation_repositories.removed`

**Review by rachellerathbone — CHANGES_REQUESTED** [review]

**Comment by KAbakumovAtl** [comment]
@rachellerathbone just made a PR for adding metrics https://github.com/atlassian/github-for-jira/pull/761 so we need to add metrics to the delete endpoint here if Rachelle merge it first

File: app.yml (hunk 1 above)
Comment (atrigueiro) [reply app.yml:43]: Nope, we're not adding a new permission, just subscribing to a new event.

**Review by rachellerathbone — APPROVED** [review]

File: src/github/index.ts (hunk 2 above)
Comment (atrigueiro) [reply src/github/index.ts:59]: Yeah you'll need to make your local app listen to the `repository` event now and that should work.

**Pull request merged**